Add cardinaldash to report data back to laptop

colescott opened the pull request.

colescott committed:
Add basic cardinaldash code
- Add dashboard class for sending data
- Add PDP class for sending PDP data
- Add VictorSP class for sending data about the victor state

colescott committed:
Update dashboard to send data and be less jank
- Add websocket data sending to server via mDNS
- Change Dashboard.cpp to use less janky storage
  - Change to use one vector for both types of values
  - Change subscribe methods to call base method
- Fix deploy failure from incorrect target mDNS address
- Add sysProps to gitignore

colescott committed:
Fix crash when starting robot code related to improper handling of
pointers/references

colescott committed:
Move VictorSP constructor from h to cpp file

colescott committed:
Add IterativeRobot class for Robot class to extend
- IterativeRobot now handles init and periodic functions
- Server now handles jsonification

colescott committed:
Fix console spam from easywsclient failing to connect

colescott committed:
Switch lib to websocketpp server mode

This change it needed because of the switch to the robot acting as a server.
Switch from client to server allows multiple clients to connect to the robot and track data
It also fixes the unknown ip issue regarding the driver station.
A few additional changes include closing the port after the program gets a SIGTERM and
adding a IterativeRobot class

colescott committed:
Add command class to cardinaldash

- Fix gitignore
- Fix some messages and stuff

colescott committed:
Merge branch 'master' into cardinaldash

colescott committed:
Lint files

erichkleung commented on src/CardinalDash/Dashboard.cpp:
```diff
@@ -0,0 +1,181 @@
+        SubscribeToValue ( name, Dashboard::DataType::INT, ( void* ) callback, object );
+    }
+
+// Well aren't we nosy?
```
no, just thorough

erichkleung commented on src/CardinalDash/Server.cpp:
```diff
@@ -0,0 +1,99 @@
+
+    void Server::RunServer()
+    {
+        m_server.listen ( 5800 );
```
pull out the int into config or at least an int at the top of the file called PORT or something

erichkleung commented on src/Robot.cpp:
```diff
@@ -1,87 +1,79 @@
-	 */
-	void DisabledInit() override {
+    void RobotPeriodic() override {
+        IterativeRobot::RobotPeriodic();
```
wanna use super/inherited? http://stackoverflow.com/questions/180601/using-super-in-c

colescott committed:
Fix minor things

- Add port const in Server.h
- Move robot class declaration to header file
- Fix some issues with broken stuffs cause im bad at stash

colescott committed:
Add port to server header for real this time

colescott committed:
Add super typedef to robot

This is fun :D
